### PR TITLE
Update | Add DAO registered name in favorites

### LINKED ISSUE #683: Bug: Use DAO name or ENS in Favorites list
### Overview

We're only showing the truncated address in Favorites.  The order we should use is:

1. ENS
2. DAO Name
3. Address (if for some reason there's no name, which shouldn't happen)

### Steps to reproduce

1. Favorite a DAO with a string name or ENS
2. View the Favorites list

### Screenshots / Video

Screen Shot 2022-11-18 at 4 22 41 PM

### Version

Dev (app.dev.fractalframework.xyz/)

### Browser

All

### Operating System

_No response_

### web3 wallet

_No response_

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
Updates Favorites to use DAO registered name if available

## Notes
Looking closer at the ENS name implementation as stated in the issue. it looks like even before this update this should have been working for the 'usuldao.eth' as setup. I've found some additonal information about it here [ENS DOC](https://docs.ens.domains/dapp-developer-guide/resolving-names) under the section 'Reverse Resolution'

I've updated `useDAOName` to include ens lookup in the proper order

## Issue / Notion doc (if applicable)
closes #683 

## Screenshots (if applicable)
image

### CODE REVIEW

File: src/hooks/DAO/useDAOName.ts [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
-  const { displayName } = useDisplayName(address);
   const [daoRegistryName, setDAORegistryName] = useState<string>();
+  const ensName = useENSName(address);
 
   const getDaoName = useCallback(async () => {
     if (!fractalNameRegistryContract || !address) {
       return '';
     }
+
     const events = await fractalNameRegistryContract.queryFilter(
```
Comment (herbig) [inline src/hooks/DAO/useDAOName.ts:19]: do we need to wait here if we already have an `ensName`?

**Comment by herbig** [comment]
@Da-Colon should the ENS lookup work for this PR? The screenshot you gave still shows usuldao.eth appearing as its eth address instead

**Comment by Da-Colon** [comment]
> @Da-Colon should the ENS lookup work for this PR? The screenshot you gave still shows usuldao.eth appearing as its eth address instead

This pertains to the conversation last night. As far as I understand this is working properly on our side. There is something on the ENS side that must be done to resolve the address to the name properly

**Comment by herbig** [comment]
> > @Da-Colon should the ENS lookup work for this PR? The screenshot you gave still shows usuldao.eth appearing as its eth address instead
> 
> This pertains to the conversation last night. As far as I understand this is working properly on our side. There is something on the ENS side that must be done to resolve the address to the name properly

ok, cool

How does the lookup work in the search bar?

**Review by herbig — APPROVED** [review]

**Comment by Da-Colon** [comment]
Resolving name 'usuldao.eth' -> daoAddress is done by using the `resolveName` method see `useAddress` hook. according to the docs resolving the name in this direction is safe. but address -> name has to be more secure, so there is an extra step that must be taken if the controller is not === to the address. as far as my understanding

File: src/hooks/DAO/useDAOName.ts [hunk 2]
```diff
@@ -1,29 +1,44 @@
...
 
     const latestEvent = events[0];
     if (!latestEvent) {
-      setDAORegistryName(displayName);
+      if (!address) {
+        setDAORegistryName('');
+        return;
+      }
+
+      setDAORegistryName(createAccountSubstring(address));
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/hooks/DAO/useDAOName.ts:35]: One argument for keeeping useDisplayName is that not all addresses need to be checked against DAO names? other wise yeah a PR can be created to replaces useDisplayNames with useDAOName

[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/hooks/DAO/useDAOName.ts:35]: yessir, so the reason I had to extract that bit of code is for the order of how we want the names to take over. ENS should take priority followed by registered DAO name and then subString.

using `useDisplayNames` within the `useENSName` hook was causing a different order. registered DAO -> ENS -> Substring.

**Pull request merged**